R/add_weights_cluster.R

```diff
@@ -119,19 +119,86 @@ add_weights_cluster <- function(x, cl,
-  indiv_surv <- tidyselect::vars_select(colnames(x),  {{interviewed}})
+  clus_id_cl <- tidyselect::vars_select(colnames(cl), {{cluster_cl}}, .strict = FALSE)
+  hh_id_cl   <- tidyselect::vars_select(colnames(cl), {{household_cl}}, .strict = FALSE)
+  clus_id_x  <- tidyselect::vars_select(colnames(x),  {{cluster_x}}, .strict = FALSE)
+  hh_id_x    <- tidyselect::vars_select(colnames(x),  {{household_x}}, .strict = FALSE)
+  indiv_eli  <- tidyselect::vars_select(colnames(x),  {{eligible}}, .strict = FALSE)
+  indiv_surv <- tidyselect::vars_select(colnames(x),  {{interviewed}}, .strict = FALSE)
+
+  ## checks (errors / warnings)
+
+  # check if variables are in datasets (spelling) - throws error
+
+  # for cluster dataset
+  if (length(clus_id_cl) == 0) {
+    # pull the environment from which this was called (i.e. the function)
+    cll   <- match.call()
+
+    # put the variable of interest in quotations
+    ppltn <- rlang::as_name(rlang::enquo(cluster_cl))
+    # throw error saying which variable not in which dataset
+    stop(glue::glue("{ppltn} is not one of the columns in {cll[['cl']]}, check spelling"))
+  }
+
+  if (length(hh_id_cl) == 0) {
+    cll   <- match.call()
+    ppltn <- rlang::as_name(rlang::enquo(household_cl))
+    stop(glue::glue("{ppltn} is not one of the columns in {cll[['cl']]}, check spelling"))
+  }
+
+  # for study dataset
+  if (length(clus_id_x) == 0) {
+    cll   <- match.call()
+    ppltn <- rlang::as_name(rlang::enquo(cluster_x))
+    stop(glue::glue("{ppltn} is not one of the columns in {cll[['x']]}, check spelling"))
+  }
+
+  if (length(hh_id_x) == 0) {
+    cll   <- match.call()
+    ppltn <- rlang::as_name(rlang::enquo(household_x))
+    stop(glue::glue("{ppltn} is not one of the columns in {cll[['x']]}, check spelling"))
+  }
+
+  if (length(indiv_eli) == 0) {
+    cll   <- match.call()
+    ppltn <- rlang::as_name(rlang::enquo(eligible))
+    stop(glue::glue("{ppltn} is not one of the columns in {cll[['x']]}, check spelling"))
+  }
+
+  if (length(indiv_surv) == 0) {
+    cll   <- match.call()
+    ppltn <- rlang::as_name(rlang::enquo(interviewed))
+    stop(glue::glue("{ppltn} is not one of the columns in {cll[['x']]}, check spelling"))
+  }
+
+
+  # check if there are duplicate cluster names (i.e. counted twice) - throws error
+  dup_clus <- cl[duplicated(cl[ , clus_id_cl]) , clus_id_cl]
+
+  if (length(dup_clus) > 0) {
+    cll   <- match.call()
+    stop(glue::glue("Cluster names are duplicated in {clus_id_cl} variable of {cll[['cl']]} dataset"))
+  }
+
+  # check how many cluster names in the study data match cluster counts data
+  matching_clusters <- unique(x[[clus_id_x]][which(!x[[clus_id_x]] %in% cl[[clus_id_cl]])])
+
+  if (length(matching_clusters) > 0) {
+    cll <- match.call()
+    warning(glue::glue("The following cluster names in the {clus_id_x} variable are not in the {cll[['cl']]} dataset: ",
+                       glue::glue_collapse(matching_clusters, sep = ",")))
+  }
```

Comment:
Breaking this up and renaming for clarity
```suggestion
  non_matching_clusters <- !x[[clus_id_x]] %in% cl[[clus_id_cl]]

  if (any(non_matching_clusters)) {
    nm_cluster_names <- unique(x[[clus_id_x]][non_matching_clusters])
    cll <- match.call()
    warning(glue::glue("The following cluster names in the {clus_id_x} variable are not in the {cll[['cl']]} dataset: ",
                       glue::glue_collapse(nm_cluster_names, sep = ",")))
  }
```